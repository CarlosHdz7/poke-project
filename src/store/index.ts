/* eslint-disable no-underscore-dangle */
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../rootReducer';

const store = createStore(
  rootReducer,
  compose(
    applyMiddleware(thunk),
    (window as any).__REDUX_DEVTOOLS_EXTENSION__ ? (window as any).__REDUX_DEVTOOLS_EXTENSION__() : (f: any) => f,
  ),
);

export default store;
