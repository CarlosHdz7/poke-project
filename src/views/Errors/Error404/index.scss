@import 'styles/settings.scss';

.error-container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.error-title {
  font-size: 100px;
  margin-bottom: 5px;
  color: $wet-asphalt;
}

.error-subtitle {
  margin-top: 0px;
  margin-bottom: 30px;
  font-weight: 700;
  color: $wet-asphalt;
}

.error-button {
  padding: 10px;
  font-size: 16px;
  color: $white;
  border: none;
  border-radius: 8px;
  background-color: $wisteria;
  cursor: pointer;
}
