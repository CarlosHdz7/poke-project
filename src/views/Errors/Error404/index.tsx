import React from 'react';
import { useNavigate } from 'react-router-dom';

import './index.scss';

const Error404 = () => {
  const navigate = useNavigate();

  const goToList = () => {
    navigate('/pokemons');
  };
  return (
    <div className="error-container">
      <p className="error-title">404</p>
      <p className="error-subtitle">Pokemon not found</p>
      <button type="button" className="error-button" onClick={goToList}>
        Go back to list
      </button>
    </div>
  );
};

export default Error404;
