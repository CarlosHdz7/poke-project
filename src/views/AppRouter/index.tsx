import React from 'react';
import { Route, Routes } from 'react-router-dom';

import Navbar from 'components/layout/Navbar';
import AboutView from 'views/AboutView';
import BookmarksView from 'views/BookmarksView';
import PokemonDetailView from 'views/PokemonDetailView';
import PokemonsView from 'views/PokemonsView';
import Error404 from 'views/Errors/Error404';
import appRoutes from 'routes';

const AppRouter = () => (
  <>
    <Navbar />
    <div className="container">
      <Routes>
        <Route path={appRoutes.POKEMONS} element={<PokemonsView />} />
        <Route path={appRoutes.POKEMON_DETAIL} element={<PokemonDetailView />} />
        <Route path={appRoutes.BOOKMARKS} element={<BookmarksView />} />
        <Route path={appRoutes.ABOUT} element={<AboutView />} />
        <Route path={appRoutes.ERROR_404} element={<Error404 />} />
        <Route path={appRoutes.HOME} element={<PokemonsView />} />
      </Routes>
    </div>
  </>
);

export default AppRouter;
