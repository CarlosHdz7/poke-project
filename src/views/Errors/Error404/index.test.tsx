import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import store from 'store';
import Error404 from '.';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';
import AppRouter from 'views/AppRouter';

describe('Testing 404 page', () => {
  it('should render correctly', async () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Error404 />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should show the correct text', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Error404 />
        </BrowserRouter>
      </Provider>,
    );

    expect(screen.queryByText(/404/i)).not.toBe(null);
    expect(screen.queryByText(/Pokemon not found/i)).not.toBe(null);
    expect(screen.queryByText(/Go back to list/i)).not.toBe(null);
  });

  it('should redirect to pokemons list', async () => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <Error404 />
          <AppRouter />
        </BrowserRouter>
      </Provider>,
    );
    const button = screen.getByText(/Go back to list/i);
    userEvent.click(button);

    await waitFor(() => {
      expect(screen.queryByText('Pokemons List')).toBeInTheDocument();
    });
  });
});
