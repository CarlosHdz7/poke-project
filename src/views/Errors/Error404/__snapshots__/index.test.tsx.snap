// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing 404 page should render correctly 1`] = `
<div
  className="error-container"
>
  <p
    className="error-title"
  >
    404
  </p>
  <p
    className="error-subtitle"
  >
    Pokemon not found
  </p>
  <button
    className="error-button"
    onClick={[Function]}
    type="button"
  >
    Go back to list
  </button>
</div>
`;
